### ISSUE TITLE
Wrong quantiles calculated for some Truncated

### ISSUE THREAD

**Issue by waldie11** [JuliaStats/Distributions.jl#1726]
Conform/Expected behavior:
```
julia > using Distributions
julia > quantile( truncated(Normal(),0,1), 1)
1.0 # == d.upper
```

Faulty behavior though there is better information available within `Truncated.d.upper` 
```
julia > quantile( truncated(Normal(0.1,0.01),0,1), 1)
Inf # == quantile(Normal(), 1)
```
Another apparent rounding error example
```
julia > quantile(truncated(Normal(-0.1,2),0,1),1)
1.0000000000000002 # still bigger than expected 1.0
```

Researching the line in `src/truncate.jl`
```
quantile(d::Truncated, p::Real) = quantile(d.untruncated, d.lcdf + p * d.tp)
```
offers more insight: it suffers from non-neglictible rounding errors in the sum of  `d.lcdf+d.tp`$\rightarrow 1$ as $p=1$

Iam suggesting a fix as in #1727 

Affected versions are probably all since `@truncate` macro was [introduced](https://github.com/JuliaStats/Distributions.jl/commit/06a268405f49b648665027c85ad75d26100fad47) . It can be reproduced at least on current master branch, 0.25.64 and 0.25.94

**Comment by waldie11** [comment]
fixed by #1727

**Comment by waldie11** [comment]
fixed by #1727
